### ISSUE TITLE
Adguard works only in wi-fi mode

### ISSUE THREAD

**Issue by vozersky** [AdguardTeam/AdguardForAndroid#425]
log's attached :[adguard.zip](https://github.com/AdguardTeam/AdguardForAndroid/files/202981/adguard.zip)

Problems with Adguard while using mobile network. Filtering doesn't work. But sometimes it works if you refresh the same webpage several times.

OLP-146-82501

**Comment by ameshkov** [comment]
There is not much info about user's configuration.

Here are some questions about it:
1. Ask user to examine the APN settings. Has he a proxy server set there? Could be the same issue as in #432 
2. I'd like to see the whole user's configuration. Please ask him to send a message with debug info attached (configuration info will be also attached in this case).

**Comment by Dodegun** [comment]
hello. I attached my APN Settings. It is fixed by mobile service provider(SKT)
![screenshot_20160411-102134](https://cloud.githubusercontent.com/assets/18387079/14414640/f5d68cbc-ffd0-11e5-97f1-e699c6141d27.jpg)

and what do you want know my configuration? I already tried "Send detailed system info" in adguard app, but it is always failed.
I'm using SM-G925S(Galaxy S6 edge), Android Version is 6.0.1, no rooted.

If you tell me How can I get the debug info, I'll do it.

**Comment by ameshkov** [comment]
@Dodegun 

> but it is always failed.
Yep, I see what's the problem in your log file: https://github.com/AdguardTeam/AdguardForAndroid/issues/439


I've examined the log and you know, it seems that ads are filtered. Maybe some connections aren't filtered at all and that's why I see nothing in the log?

How do you check that ad blocking works? Do you open some particular website?

**Comment by vozersky** [comment]
@Dodegun  try this new build : https://dl.dropboxusercontent.com/u/57517781/adguard-android-2.5.85.apk

You should at least be able to send "detailed system info" through Adguard interface with it.
Please tell me how it goes.

**Comment by Dodegun** [comment]
@ameshkov 
This problem is occured every site.
I usually check 'tcafeaa.com' and 'ruliweb.com'. there are big banner top and left side.

@vozersky 
ok. I submit new ticket using 2.5.85. there are debug/state and new adguard log.
#BKA-665-75992

**Comment by vozersky** [comment]
new log and debug from #BKA-665-75992  [logs.zip](https://github.com/AdguardTeam/AdguardForAndroid/files/216827/logs.zip)

**Comment by ameshkov** [comment]
I still see that http://ruliweb.daum.net/ is filtered and some ads are blocked (just don't know about these specific banners). The same for http://tcafeaa.com/.

I wonder could it be just a filters issue. For instance they serve a different ad for mobile users?

@Dodegun one more thing, I see both banners on http://tcafeaa.com/, these are just not blocked by the filters you use.

Try adding these rules to your user filter:
```
tcafeaa.com###divpop
tcafeaa.com##a[href^="http://goo.gl"] > img
```

Does it help?

**Comment by Dodegun** [comment]
both site did not support mobile view. I think it is same ad.
I tested your filter. there are still problem.

I tested several times with only five filter for tcafeaa.com and ruliweb.daum.net.

> tcafeaa.com###divpop
> tcafeaa.com###s_word
> tcafeaa.com##a[href^="http://goo.gl"] > img
> tcafeaa.com##a[href^="http://www.totoro"] > img
> /display.ad.(for ruliweb.daum.net)

and this is the test video.
https://youtu.be/_NWDpyA3B2w

**Comment by ameshkov** [comment]
Is it possible that Samsung browser uses some kind of a data compression server when you're on mobile?

**Comment by Dodegun** [comment]
@ameshkov no. Samsung browser does not support data saver. and there are same problem with chrome browser.

**Comment by ameshkov** [comment]
@Dodegun one last question, could you please record such video along with Adguard full log?

I'd like to compare what I see in the video with what's recorded in the log.

**Comment by Dodegun** [comment]
here is another test video with full log.
Video : https://youtu.be/YqzaL7M13YQ
Log : https://drive.google.com/file/d/0ByHcIUCvSt9EY3RLQTdEakpUZDQ/view?usp=sharing

**Comment by ameshkov** [comment]
@Dodegun thank you for the log and video!

So, what do I see:
1. When you turn Wi-Fi off, Adguard does not receive a "CONNECTIVITY_CHANGE" event (that is strange). However, we receive WIFI_STATE_CHANGED event, but we usually do not react on it.
2. I see that these requests are NOT routed through AG VPN immediately after turning wi-fi off.

I suppose, that the problem is that due of missing "CONNECTIVITY_CHANGE" Adguard does not reconfigure the VPN and it stays in some incorrect state.

To check if I am right please do the same test with one change: before you open that test website go to Adguard and restart it (so that VPN will be reconfigured).

**Comment by Dodegun** [comment]
@ameshkov

turn off wifi and restart adguard - still problem
turn off wifi and phone restart - still problem

even if I never connect wifi network once , adblock is not working on mobile network.
but as i said before, adblock is working somtimes.

**Comment by Dodegun** [comment]
This problem is solved at Adguard Beta 2.8.1
adblock is working on mobile network also.

**Comment by ameshkov** [comment]
Huh, I wonder what exactly has helped..

Anyway, it's great that the issue is finally sorted out:)

**Comment by Bohdan-SUP** [comment]
@ameshkov can you check ticket 1457684, seems like a similar issue.

<details>
Phone: LG V20 (Android 7.0)
Carrier: SKT
Version: Stable/Beta (Both)
Filters: Default Filters + List-KR

- I never experienced any problem in Wifi + Adguard.
- But when I’m in LTE or 3G (any mobile connections), Adguard doesn’t block Ads.
- **‘ipv6.disable’** (in low-level settings) makes me to block Ads in mobile connections, but it also kills my push notifications.
- **‘ipv6.bypass’** also kills my push notifications, but it does nothing with Ads, unlike ‘ipv6.disable’.
- When I use adguard + ‘ipv6.disable’ in mobile, my delayed notifications comes out in full-flood when I connect to Wifi.
- I tried Opera VPN and Tunnelbear VPN app, but didn’t experienced problem about push notifications with that VPNs.
- It happens stable and latest beta versions, both.
</details>

Log - https://gist.github.com/Bohdan-SUP/4f21a92d784b410bc9bf7b9042d2795f

**Comment by ameshkov** [comment]
@Bohdan-SUP does it help to update to v2.8.76?

**Comment by vozersky** [comment]
Outdated

**Comment by vozersky** [comment]
@Bohdan-SUP we still have to ask  1457684 if it is ok after all

**Comment by nskzhd** [comment]
I'm having the same problem. Since I started to use SKT as a mobile carrier (i was using KT before and never had this issue), adblocking on mobile data hasn't been working.
I searched web and found that checking "pref.vpn.ipv6.bypass" would fix this issue, but it doesn't seem to be working. 
Is there any fix yet?

**Comment by Dodegun** [comment]
> 
> 
> @vozersky @ameshkov
> 
> I'm having the same problem. Since I started to use SKT as a mobile carrier (i was using KT before and never had this issue), adblocking on mobile data hasn't been working.
> I searched web and found that checking "pref.vpn.ipv6.bypass" would fix this issue, but it doesn't seem to be working.
> Is there any fix yet?

"pref.vpn.ipv6.bypass" and "pref.vpn.ipv6.disable" option could fix this issue for a long time. but it is not working anymore after v2.12.233 is released.  You can use v2.12.233 or latest nightly build. it works fine.

When you will use nightly build, just check "pref.vpn.ipv6.disable" only and check the filter that you want is still select well. sometimes checked filter is uncheck.

**Comment by nskzhd** [comment]
@Dodegun

Thank you so much! just installed a nightly build and cheking pref.vpn.ipv6.disable worked!
